op-batcher/batcher/channel_builder.go

```diff
@@ -256,6 +257,22 @@ func (c *channelBuilder) updateSwTimeout(batch *derive.BatchData) {
 	c.updateTimeout(timeout, ErrSeqWindowClose)
 }
 
+// updateLeaderTimeout updates the block timeout with the final block for this
+// batcher in the current leader election contract. This assumes this batcher
+// is currently the leader, based on the l1BlockNum. The timeout is only moved
+// forward if the derived timeout is earlier than the currently set timeout.
+//
+// It does nothing if the max channel duration is set to 0.
+func (c *channelBuilder) updateLeaderTimeout(lastL1Block uint64) {
+	// maybe this should be something similar to
+	// `c.updateTimeout(lastL1Block - c.cfg.SubSafetyMargin, ErrLeaderSlotWillClose)`
+	// We need to figure out what reasonable leader durations look like. I think we
+	// should have something that looks like the current epoch, with a safety margin
+	// based on being sufficiently confident that Close will be seen while
+	// isCurrentLeader still returns true. I suspect our Horizon of 10 is too small.
```

Comment:
The horizon in #72 is set to max_number_participants * number_of_slots_per_leader, would that be ok, enough?

Reply:
Oh! Yes, that'll do nicely.

Reply:
Anything greater than `number_of_slots_per_leader + 1` will work...